### PR TITLE
Implement BPS Profile deletion on tenant deletion

### LINKED ISSUE #49: Implement deletion of BPS Profiles on tenant deletion process
**Description:**
The `WorkflowImplTenantMgtListener` is fired on tenant deletion process. With that, the BPS Profiles created under that tenant should get deleted.

**Related Issues:**
wso2/carbon-multitenancy#146

### PR DESCRIPTION
**Pull request by sumedhe**
### Proposed changes in this pull request
Implement deletion of BPS Profiles when a Tenant gets deleted

Closes #49

### CODE REVIEW

File: components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplService.java [hunk 1]
```diff
@@ -54,6 +54,14 @@ void addBPSProfile(BPSProfile bpsProfileDTO, int tenantId)
      */
     void removeBPSProfile(String profileName) throws WorkflowImplException;
 
+    /**
+     * Remove all BPS profiles of given tenant id.
+     *
+     * @param tenantId
+     * @throws WorkflowImplException
+     */
+    void removeBPSProfiles(int tenantId) throws WorkflowImplException;
```
Comment (denuwanthi) [inline components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplService.java]: Should not we use 'default' here?
If not all the classes implementing this interface will have to implement this method.

Comment (sumedhe) [reply components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplService.java]: Agree!

**Review by omindu — DISMISSED** [review]

**Review by IsuraD — DISMISSED** [review]

File: components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplServiceImpl.java [hunk 2]
```diff
@@ -134,6 +134,29 @@ public void removeBPSProfile(String profileName) throws WorkflowImplException {
 
     }
 
+    /**
+     * Remove all BPS Profiles of a given tenant id.
+     *
+     * @param tenantId Id of the tenant
```
Comment (omindu) [inline components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplServiceImpl.java:140]:
```suggestion
     * @param tenantId of the tenant
```

Comment (omindu) [reply components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplServiceImpl.java:140]: fix in other applicable places

File: components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplService.java [hunk 3]
```diff
@@ -54,6 +54,16 @@ void addBPSProfile(BPSProfile bpsProfileDTO, int tenantId)
      */
     void removeBPSProfile(String profileName) throws WorkflowImplException;
 
+    /**
+     * Remove all BPS profiles of given tenant id.
+     *
+     * @param tenantId
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplService.java]: Shall we complete the method comments by adding the descriptions?

Comment (somindatommy) [reply components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplService.java]: Shall we fix this in other applicable places as well?

Comment (sumedhe) [reply components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplService.java]: 👍

**Review by IsuraD — APPROVED** [review]

File: components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplService.java [hunk 4]
```diff
@@ -54,6 +54,16 @@ void addBPSProfile(BPSProfile bpsProfileDTO, int tenantId)
      */
     void removeBPSProfile(String profileName) throws WorkflowImplException;
 
+    /**
+     * Remove all BPS profiles of given tenant id.
+     *
+     * @param tenantId Id of the tenant
```
Comment (Kanapriya) [inline components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplService.java:60]: Change this as follows and check this throughout this PR
 @param tenantId of the tenant

Comment (sumedhe) [reply components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplService.java:60]: I have placed it there as, `tenantId` is the parameter name and `Id of the tenant` is the description. Hope that's fine.

**Pull request merged**